api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/ehcache/Ehcache3Properties.java

```diff
@@ -0,0 +1,144 @@
+package org.apereo.cas.configuration.model.support.ehcache;
+
+import org.apereo.cas.configuration.model.core.util.EncryptionRandomizedSigningJwtCryptographyProperties;
+import org.apereo.cas.configuration.support.RequiredProperty;
+import org.apereo.cas.configuration.support.RequiresModule;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
+
+import java.io.Serializable;
+
+/**
+ * This is {@link Ehcache3Properties}.
+ *
+ * @author Hal Deadman
+ * @since 6.2.0
+ */
+@RequiresModule(name = "cas-server-support-ehcache3-ticket-registry")
+@Getter
+@Setter
+public class Ehcache3Properties implements Serializable {
+
+    /**
+     * EhCache autocreate mode for creating config in Terracotta cluster.
+     * Creates config as specified if not present.
+     */
+    public static final String CONNECTION_MODE_AUTOCREATE = "AUTOCREATE";
+
+    /**
+     * EhCache configless mode  for creating config in Terracotta cluster.
+     * Requires config to already be defined on Terracotta cluster.
+     */
+    public static final String CONNECTION_MODE_CONFIGLESS = "CONFIGLESS";
+
+    private static final long serialVersionUID = 7772510035918976450L;
+
+    /**
+     * The name of the cache manager instance.
+     */
+    @RequiredProperty
+    private String cacheManagerName = "ticketRegistryCacheManager";
+
+    /**
+     * Builder that sets the maximum objects to be held in memory (0 = no limit).
+     */
+    private int maxElementsInMemory = 10_000;
+
+    /**
+     * Size of disk cache.
+     */
+    private int maxSizeOnDisk = 200;
+
+    /**
+     * Disk cache size units as defined by EhCache. (e.g. MB, GB)
+     */
+    private String maxSizeOnDiskUnits = "MB";
+
+    /**
+     * Per cache size of disk cache.
+     */
+    private int perCacheSizeOnDisk = 20;
+
+    /**
+     * Per cache disk cache size units as defined by EhCache. (e.g. MB, GB)
+     */
+    private String perCacheSizeOnDiskUnits = "MB";
+
+
+    /**
+     * Size of heap cache.
+     */
+    private int maxSizeOffHeap = 100;
```

Comment:
Is this supposed to be `off heap` or `of heap` ?

Reply:
It's "off heap". Ehcache has heap, offheap and disk storage with their non-clustered configuration. I think if you are pointing it at terracotta cluster then the resource pools used are all "off heap". https://stackoverflow.com/questions/6091615/difference-between-on-heap-and-off-heap